genie-core/src/main/java/com/netflix/genie/core/util/UnixProcessChecker.java

```diff
@@ -80,4 +92,32 @@ public void checkProcess() throws GenieTimeoutException, ExecuteException, IOExc
             );
         }
     }
+
+    /**
+     * Factory for {@link ProcessChecker} for UNIX systems.
```

Comment:
since it's a factory why would it be specific to UNIX systems? I'm probably missing some of your logic but I would think this would be some standalone factory class outside the scope of just the UnixProcesChecker that would create process checkers on a per OS basis. Why would we have a factory per implementation?

Reply:
Isn't that already the case?
All users are just depending on and using `ProcessChecker.Factory`.

This implementation is only able to produce UNIX process checkers, but nobody needs to know.

Reply:
Then there's no value in a factory right?

Reply:
Before we were calling `new UnixProcessChecker()`. in different parts of the code.
Now we call `ProcessChecker.Factory.get()` in those places.

We inject the factory, we configure it externally, etc. Why is this no value?

Reply:
In my opinion in this design your "factory"should be more of a factory method than a factory class cause your factory only produces one thing and is tied to the parent class it's coming from. I'd break it out into a totally separate thing but that's just me. Like i said merge it if you're comfortable with it.

Reply:
Factory is just meant to decouple and encapsulate object creation. 
Nothing says it needs to create different types of objects.

The factory interface i added is generic.
The implementation is specific but it can be swapped out, changed later, etc.